coordinator/src/notification_service.rs

```diff
@@ -0,0 +1,132 @@
+use anyhow::Context;
+use anyhow::Result;
+use std::fmt::Display;
+use tokio::sync::mpsc;
+
+/// Types of notification that can be sent to 10101 app users
+
+#[derive(Debug, Clone)]
+pub enum NotificationKind {
+    /// Coordinator would like to settle the channel
+    ChannelClose,
+    PositionSoonToExpire,
+    PositionExpire,
+}
+
+impl Display for NotificationKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NotificationKind::ChannelClose => write!(f, "ChannelClose"),
+            NotificationKind::PositionSoonToExpire => write!(f, "PositionSoonToExpire"),
+            NotificationKind::PositionExpire => write!(f, "PositionExpire"),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Notification {
+    pub user_fcm_token: String,
+    pub notification_kind: NotificationKind,
+}
+
+impl Notification {
+    pub fn new(user_fcm_token: String, notification_kind: NotificationKind) -> Self {
+        Self {
+            notification_kind,
+            user_fcm_token,
+        }
+    }
+}
+
+/// Actor managing the notifications
+pub struct NotificationService {
+    notification_sender: mpsc::Sender<Notification>,
+}
+
+impl NotificationService {
+    /// Start the notification service
+    ///
+    /// If on empty string is passed in the constructor, the service will not send any notification.
+    /// It will only log the notification that it would have sent.
+    pub fn new(fcm_api_key: String) -> Self {
+        if fcm_api_key.is_empty() {
+            // Lot it as error, as it production it should always be set
+            tracing::error!("FCM API key is empty. No notifications will not be sent.");
+        }
+
+        let (notification_sender, mut notification_receiver) = mpsc::channel(100);
+
+        // TODO: use RAII here
+        tokio::spawn({
+            let fcm_api_key = fcm_api_key;
+            let client = fcm::Client::new();
+            async move {
+                while let Some(Notification {
+                    user_fcm_token,
+                    notification_kind,
+                }) = notification_receiver.recv().await
+                {
+                    tracing::info!("Sending notification: {notification_kind} to {user_fcm_token}");
```

Comment:
Imho it's nicer to _annotate_  (not sure if this is the right term) the log message with the variables, i.e. 

```
tracing::info!(notification_kind, user_fcm_token, "Sending notification");
```